src/main/kotlin/net/maizegenetics/phgv2/utils/GeneralUtilities.kt

```diff
@@ -29,3 +34,149 @@ fun inputStreamProcessing(tiledbList: BufferedInputStream): List<String> {
+ * For merged ranges, the gene string is appended to the existing gene string.
+ */
+fun addRange(geneRange: RangeMap<Position, String>, range: Range<Position>, gene:String) {
+    val overlaps: List<Map.Entry<Range<Position>, String>> =
+        ArrayList<Map.Entry<Range<Position>, String>>(
+            geneRange.subRangeMap(range).asMapOfRanges().entries
+        )
+
+    //if overlaps has length, merge ranges together
+    if (overlaps.size != 0) {
+        val overlappingEntry: Map.Entry<Range<Position>, String>? = geneRange.getEntry(
+            overlaps[0].key.lowerEndpoint()
+        )
+        //then use the combined range and assign the call
+        val newGene = "${overlappingEntry!!.value}-$gene"
+
+        // Update overlappingEntry value with new merged gene value.
+        // 2nd put is to ensure new entry is merged with the new value
+        geneRange.put(overlappingEntry.key, newGene)
+        geneRange.putCoalescing(range, newGene)
+    } else {
+        geneRange.put(range, gene)
+    }
+}
+
+/**
+ * This functions takes a RangeMap of Positions, plus a "pad" value, and returns a new RangeMap with the
+ * ranges expanded by the pad value.  The reference sequence is used to determine chromosome sizes.
+ * If there are not enough bases between ranges to add the specified padding, the existing base pairs
+ * are split between the ranges, and padding is cut to what is available.  If the padding will exceed
+ * the end of the chromsome, it is truncated as appropriate.
+ */
+fun createFlankingList(geneRange:RangeMap<Position,String>, numFlanking:Int, refSeq: Map<String, NucSeq>): RangeMap<Position,String> {
+
+    // there are no embedded or overlapped entries in the geneRange map.
+    // The goal is to add "numFlanking" flanking to start/end of each entry.
+    // There may not be 2*numFlanking between each gene entry.  If not, split the difference
+    // between:  half to geneA end, half to geneB start.  This is done in
+    // findFlankingStartPos() and findFlankingEndPos()
+    val flankingRange: RangeMap<Position, String> = TreeRangeMap.create()
+
+    try {
+        geneRange.asMapOfRanges().entries.forEach { range ->
+            val data:Range<Position> = range.key
+            val chr = data.lowerEndpoint().contig
+            if (!(refSeq.keys.contains(chr))) {
+                throw IllegalArgumentException("createFlankingList: chrom $chr not found in reference fasta.")
+            }
+            val chrLen = refSeq[chr]!!.size()
+
+            // Find new start/end positions with specified number of flanking, add this position to the map
+            var flankingStart = findFlankingStartPos(geneRange,data, numFlanking)
+            // adjustment for 0-based occurred when creating GeneRanges
+            flankingStart = Position(flankingStart.contig,flankingStart.position) // start was adjusted to be 0-based in geneRange
```

Comment:
This line overwrites the value on line 92?

Reply:
Line 94 should be removed - I was initially doing 0-based adjustment here vs earlier and needed line 94, but now it just copies the same position.  I'll delete.